scripts/travis/permissions-workaround.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ "$(whoami)" != 'root' ]; then
+    echo 'Please run with sudo'
+    exit 2
+fi
+
+UBUNTU_VERSION=`lsb_release -a 2>&1 | grep Codename | grep -v "LSB" | awk '{print $2}'`
+
+# Workaround for Travis on Ubuntu Xenial so local runner integration tests work
+# when executing them under user "stanley" (by default Travis checks out the
+# code and runs tests under a different system user).
+# NOTE: We need to pass "--exe" flag to nosetests when using this workaround.
+if [ "${UBUNTU_VERSION}" == "xenial" ]; then
+  echo "Applying workaround for stanley user permissions issue to /home/travis on Xenial"
+  chmod 777 -R /home/travis
```

Comment:
Using octal notation here forces all files to be executable, which then forces you to modify a million files adding the `--exe` flag to all `nosetests` invocations.

Instead, I would use the more human-friendly `chmod` syntax to make only the permissions changes that you want to:

```bash
chmod ugo+r -R /home/travis
```

Also consider that you can change the owner of files to `stanley` instead of changing permissions:

```bash
chown -R stanley /home/travis/...
```

Or better yet, we should figure out if we can get Travis to run setup scripts as the `stanley` user.

Reply:
yes. this is very ugly. switching to `ugo+r` would be much nicer.

Reply:
I used `a+rwX`. Seems to work. Maybe figuring out how to run the tests as stanley would be good in a different PR. I'm updating mongo/rabbitmq repos/versions to make the tests work in the new environment, but I'm avoiding making any other changes.

Reply:
Why are you touching the executable bit at all though? If you use the `+` notation, you don't have to specify it and `chmod` will leave the executable bit alone.

I'm also not a huge fan of changing the permissions for the owner, the group, and all others. That's very likely not how ST2 is run in production systems, and it shouldn't be how it's run during tests.

Would it be simpler to just `chown` the ST2 code to the `stanley` user?